examples/plugins/engine/examplePluginEngineRead.cpp

```diff
@@ -82,30 +89,34 @@ int main(int argc, char *argv[])
         }
         adios2::Engine reader = io.Open("TestPlugin", adios2::Mode::Read);
 
-        auto var = io.InquireVariable<float>("data");
-        if (!var)
-        {
-            std::cout << "variable does not exist" << std::endl;
-        }
-
         if (streaming)
         {
-            testStreaming(reader, myFloats, var);
+            testStreaming(io, reader, myFloats);
         }
         else
         {
-            std::vector<float> readFloats;
-            reader.Get<float>(var, readFloats);
-            reader.PerformGets();
-
-            if (readFloats == myFloats)
+            reader.BeginStep();
+            auto var = io.InquireVariable<float>("data");
+            if (!var)
             {
-                std::cout << "data was read correctly!" << std::endl;
+                std::cout << "variable does not exist" << std::endl;
             }
             else
             {
-                std::cout << "data was not read correctly!" << std::endl;
+                std::vector<float> readFloats;
+                reader.Get<float>(var, readFloats);
+                reader.PerformGets();
+
+                if (readFloats == myFloats)
+                {
+                    std::cout << "data was read correctly!" << std::endl;
+                }
+                else
+                {
+                    std::cout << "data was not read correctly!" << std::endl;
+                }
             }
+            reader.EndStep();
```

Comment:
Maybe this part should be removed? I think when I originally wrote it my thinking was to show using it with Begin/EndStep (in the testStreaming function) and then to show that it works with just using PerformGets and not using Begin/EndStep, but it should probably just be simplified to only do what's in the testStreaming function.

Reply:
Fixed